universal-login-contracts/test/contracts/ProxyMasterCopy.js

```diff
@@ -31,38 +29,55 @@ describe('ProxyMasterCopy', async () => {
   let publicKey;
   let mockToken;
   let wallet;
-  let msg;
-  let signature;
+  let changeMasterCopyFunc;
+  let data;
   let anotherWallet;
-  let invalidSignature;
-  let relayerBalance;
-  let relayerTokenBalance;
 
   beforeEach(async () => {
     ({provider, masterCopy, proxy, privateKey, publicKey, mockToken, wallet} = await loadFixture(basicMasterAndProxy));
-    msg = {...transferMessage, from: proxy.address};
-    signature = calculateMessageSignature(privateKey, msg);
+    changeMasterCopyFunc = new utils.Interface(MockMasterCopy.interface).functions.changeMasterCopy;
     [anotherWallet] = await getWallets(provider);
-    invalidSignature = calculateMessageSignature(anotherWallet.privateKey, msg);
-    relayerBalance = await wallet.getBalance();
-    relayerTokenBalance = await mockToken.balanceOf(wallet.address);
   });
 
   describe('MasterCopy', async () => {
-    xit('should fail calling authorized function directly');
+    it('should fail if authorized function is called directly', async () => {
+      await expect(masterCopy.changeMasterCopy(to1)).to.be.revertedWith('Method can only be called from this contract');
+    });
   });
 
   describe('Proxy', async () => {
     it('should be properly constructed', async () => {
       expect(await proxy.implementation()).to.eq(masterCopy.address);
     });
 
-    xit('should fail if function does not exist in MasterCopy');
+    it('should fail if function does not exist in MasterCopy', async () => {
+      await expect(wallet.sendTransaction({to: proxy.address, data: [], gasPrice, gasLimit})).to.be.reverted;
+    });
 
-    xit('should call function in MasterCopy');
+    xit('should fail if calling changeMasterCopy in MasterCopy with zero address', async () => {
+      // data = changeMasterCopyFunc.encode([utils.formatBytes32String(0)]);
+      data = '0x7de7edef0000000000000000000000000000000000000000000000000000000000000000';
```

Comment:
Hard-coded this way just because the commented line above fast-fails in ethers without reaching the `MasterCopy` contract